### PR TITLE
[llvm-lib] Handle MIPS architecture

### PR DESCRIPTION
**Pull request by hpoussin**
- add a test to check values for /machine argument
- add a test to check if machine is correctly inferred from inputs

### CODE REVIEW

File: llvm/lib/Object/WindowsMachineFlag.cpp [hunk 1]
```diff
@@ -28,6 +28,7 @@ COFF::MachineTypes llvm::getMachineType(StringRef S) {
       .Case("arm64", COFF::IMAGE_FILE_MACHINE_ARM64)
       .Case("arm64ec", COFF::IMAGE_FILE_MACHINE_ARM64EC)
       .Case("arm64x", COFF::IMAGE_FILE_MACHINE_ARM64X)
+      .Case("mips", COFF::IMAGE_FILE_MACHINE_R4000)
```
Comment (wzssyqa) [inline llvm/lib/Object/WindowsMachineFlag.cpp:31]: Should it be `mipsel`?

File: llvm/test/tools/llvm-lib/machine-opt.test [hunk 2]
```diff
@@ -0,0 +1,13 @@
+RUN: rm -f %t.lib
+
+RUN: llvm-lib /out:%t.lib /machine:i386 2>&1 | FileCheck --check-prefix=EMPTYWARN %s
+RUN: llvm-lib /out:%t.lib /machine:amd64 2>&1 | FileCheck --check-prefix=EMPTYWARN %s
+
+RUN: llvm-lib /out:%t.lib /machine:mips 2>&1 | FileCheck --check-prefix=EMPTYWARN %s
```
Comment (wzssyqa) [inline llvm/test/tools/llvm-lib/machine-opt.test:6]: mipsel?

File: llvm/lib/Object/WindowsMachineFlag.cpp (hunk 1 above)
Comment (hpoussin) [reply llvm/lib/Object/WindowsMachineFlag.cpp:31]: No, it must be `mips` to be compatible with Microsoft lib.exe

File: llvm/test/tools/llvm-lib/machine-opt.test (hunk 2 above)
Comment (hpoussin) [reply llvm/test/tools/llvm-lib/machine-opt.test:6]: No, it must be `mips` to be compatible with Microsoft lib.exe

File: llvm/lib/Object/WindowsMachineFlag.cpp (hunk 1 above)
Comment (wzssyqa) [reply llvm/lib/Object/WindowsMachineFlag.cpp:31]: Can you add a comment here?

Comment (hpoussin) [reply llvm/lib/Object/WindowsMachineFlag.cpp:32]: Done

**Comment by wzssyqa** [comment]
I reverted this commit due to failures from CI.
Can you have a look at what happened and resubmit this patch?

**Pull request merged**